src/workflow/StreamlitUI.py

```diff
@@ -102,7 +100,9 @@ def upload_widget(
 
         # Local file upload option: via directory path
         if st.session_state.location == "local":
-            c2.markdown("**OR copy files from local folder**")
+            c2_text, c2_checkbox = c2.columns([2, 1])
+            c2_text.markdown("**OR copy files from local folder**")
+            use_symlink = c2_checkbox.checkbox("Use symlinks", key=f"{key}-symlink", help="Use symbolic links instead of copying files.")
```

Comment:
I added a checkbox widget to allow the user to toggle on or off whether they want to use symbolic links or not. The Default value for the checkbox is False. I added a streamlit warning message at the end of the form as well if symlink is turned on to inform the user.

Reply:
I actually like this solution a lot as this makes the user less dependent on developers implementing symlinks into their WebApps (which is extra work and in a lot of cases likely not the highest priority). 

I think it would be good to support both approaches. Enabling developers to use symlinks without any user action, while also enabling advanced users to use symlinks for all input files.